gradle-failure-reports-exceptions/src/main/java/com/palantir/gradle/failurereports/exceptions/MinimalException.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.failurereports.exceptions;
+
+/**
+ * An exception type that simply surfaces the given exception message and does not assume the exception
+ * has a useful stacktrace, and thus is useful for surfacing errors derived from a subprocess.
```

Comment:
I think we're still going to want to have a stacktrace for any failed task. It's still useful to know which piece of code called the external process, and how we got there. So long as the stacktrace is at the bottom of the message, surely that is enough?

Reply:
I disagree for this case, where specifically this is used with better-exec because this error is thrown only when a better-exec action fails so you know what command failed and how, and so I want to show the minimal information necessary to remediate the failure, which is just the hatch output in this case. The causal chain and full stacktrace don't provide any useful information apart from that this error originates from better-exec, for example this is the causal chain:

> org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':functions-python-api:hatchCheckFormat'.
> 	org.gradle.workers.internal.DefaultWorkerExecutor.WorkExecutionException: A failure occurred while executing com.palantir.gradle.betterexec.BetterExecAction
> 	java.lang.RuntimeException: Task failed after 1 attempts with exit code 1. See output at: https://circle.palantir.build/gh/foundry/functions-python-api/3062/artifacts/0/~/artifacts/functions-python-api.hatchCheckFormat.log

and the full stacktrace is similar, just way more verbose. Additionally, printing the causal chain and the stacktrace is what prints the hatch output multiple times, so originally it was printing three times and this was something I wanted to explicitly avoid. Lastly, if you want to see the stacktraces they are still present in the full logs.

Reply:
**I disagree for this case, where specifically this is used with better-exec because this error is thrown only when a better-exec action fails so you know what command failed and how, and so I want to show the minimal information necessary to remediate the failure, which is just the hatch output in this case. The causal chain and full stacktrace don't provide any useful information apart from that this error originates from better-exec, for example this is the causal chain:

> org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':functions-python-api:hatchCheckFormat'.
> 	org.gradle.workers.internal.DefaultWorkerExecutor.WorkExecutionException: A failure occurred while executing com.palantir.gradle.betterexec.BetterExecAction
> 	java.lang.RuntimeException: Task failed after 1 attempts with exit code 1. See output at:

and the full stacktrace is similar, just way more verbose. Additionally, printing the causal chain and the stacktrace is what prints the hatch output multiple times, so originally it was printing three times and this was something I wanted to explicitly avoid. Lastly, if you want to see the stacktraces they are still present in the full logs.**

Reply:
Yeah, I guess for someone who needs to debug the actual gradle parts, all that will still be accessible.